wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -299,14 +303,28 @@ func (v *Wrapper) getKeyVaultClient(withCertPool *x509.CertPool) (*azkeys.Client
 			return nil, fmt.Errorf("failed to get client secret credentials %w", err)
 		}
 	case v.clientID != "":
-		// Try a managed service credential with a specified client id
-		clientID := azidentity.ClientID(v.clientID)
-		cred, err = azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
-			ID: clientID,
-		})
+		// Some hoops to jump through to make sure two wrappers being setup at the same time don't step on the
+		// env var
+		managedClientIdLock.Lock()
+		oldVal, found := os.LookupEnv(EnvAzureClientId)
+		unlock := func() {
```

Comment:
any reason to not use a `defer`?

Reply:
Yeah, the defer takes the execution down into the part in the code below where a connection is made, which takes several seconds sometimes.  We want to hold the lock only as long as it takes to initialize the credential provider.